storage/storage/src/main/java/one/microstream/storage/types/Destroyable.java

```diff
@@ -0,0 +1,33 @@
+package one.microstream.storage.types;
+
+/*-
+ * #%L
+ * MicroStream Storage
+ * %%
+ * Copyright (C) 2019 - 2022 MicroStream Software
+ * %%
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
```

Comment:
This is a commonly used type that should be put in the base module.
And instead of Destroyable#destroy I suggest Disposable#dispose, as used by many other frameworks.
The best package in the base module would be one.microstream.typing, where other similar types reside, like Clearable or Copyable.